modules/core/src/test/scala/com/snowplowanalytics/iglu/schemaddl/bigquery/FieldSpec.scala

```
@@ -216,4 +217,19 @@ class FieldSpec extends org.specs2.Specification { def is = s2"""
     val expected = Field("arrayTest",Type.Record(List(Field("imp",Type.String,Mode.Repeated))),Mode.Required)
     Field.build("arrayTest", input, true) must beEqualTo(expected)
   }
+
+  def e9 = {
+    (fieldNormalName("") must beEqualTo("")) and
+      (fieldNormalName("test1") must beEqualTo("test1")) and
+      (fieldNormalName("test1Test2Test3") must beEqualTo("test1_test2_test3")) and
+      (fieldNormalName("Test1Test2TEST3") must beEqualTo("test1_test2_test3")) and
+      (fieldNormalName("test1,test2.test3;test4") must beEqualTo("test1_test2_test3_test4")) and
+      (fieldNormalName("1test1,test2.test3;test4") must beEqualTo("_1test1_test2_test3_test4")) and
+      (fieldNormalName("_50test1,test2.test3;test4") must beEqualTo("_50test1_test2_test3_test4")) and
+      (fieldNormalName("_.test1,test2.test3;test4") must beEqualTo("__test1_test2_test3_test4")) and
+      (fieldNormalName(",.;:") must beEqualTo("____")) and
```

Ugh, what happens if the schema has `,.;:` and `,,,,` columns? I'm thinking maybe if there are only unsupported characters - we actually can let it fail?

Although letting them fail still doesn't solve the proble we'd get for `,a` and `;a` columns.

---

I think that it should fail if there is any column that is not a valid BQ column name

---

I think it's pretty unlikely we'll get columns that only consist of punctuation. I can imagine `.column` or `:column`, especially if it's read from somewhere else and nobody pays attention. But `:;.,` just makes no sense. Should be fine to fail.

---

I thought about it, but it goes against our desire to treat different DBs the same. If it's a valid character in Redshift, why would it fail in BigQuery.